react/modules/tasks/tests.md

```diff
@@ -0,0 +1,189 @@
+  - Shows loading state while fetching data
+- **Data Display Tests**:
+  - Correctly displays item names and descriptions
+  - Handles missing or undefined data gracefully
+- **Error Handling Tests**:
+  - Displays error message when API call fails
+  - Shows appropriate error for different HTTP status codes (4xx, 5xx)
+
+### 3. Card/Item Component Tests
+
+- **Rendering Tests**:
+  - Displays item name and description correctly
+  - Handles missing props gracefully
+- **Interaction Tests**:
+  - Calls onClick handler when card is clicked
+  - Passes correct item data to click handler
+
+### 4. Loading Component Tests
+
+- **Rendering Tests**:
+  - Renders loading indicator (spinner, skeleton, etc.)
+  - Shows/hides based on loading prop
+- **Accessibility Tests**:
+  - Has appropriate ARIA labels for screen readers, if your loading indicator has `aria-label`
+
+### 5. Error Boundary Tests
+
+- **Error Catching Tests**:
+  - Catches and handles JavaScript errors in child components
+  - Displays fallback UI when error occurs
+  - Logs error to console
+- **Error Button Tests**:
+  - Throws error when test button is clicked
+  - Triggers error boundary fallback UI
+
+### 6. Main App Component Tests
+
+- **Integration Tests**:
+  - Makes initial API call on component mount
+  - Handles search term from localStorage on initial load
+  - Manages loading states during API calls
+- **API Integration Tests**:
+  - Calls API with correct parameters
+  - Handles successful API responses
+  - Handles API error responses
+- **State Management Tests**:
+  - Updates component state based on API responses
+  - Manages search term state correctly
+
+### 7. API Integration Tests
+
+- **Mocked API Calls**:
+  - Use `jest.mock` or `vi.mock` to mock API calls
+  - Test both success and error scenarios
+
+### 8. Setup
+
+- Lastly, update Husky to run tests on the pre-push hook, ensuring that tests are automatically executed before any code is pushed.
+
+### 9. Submission Requirements
+
+1. All tests must pass: `npm run test`
+2. Coverage report must show ≥80% in all metrics: `npm run test:coverage`
+3. No console errors or warnings during test execution
+4. Tests must be deterministic (no flaky tests)
+5. All external dependencies must be properly mocked
+
+## Points Distribution
+
+### Student can get 100 points:
+
+- **Test Runner Setup** (Jest/Vitest configured with RTL, coverage reporting works) - **20 points**
+- **Search Component Tests** (localStorage, user input, search functionality) - **15 points**
+- **Results/CardList Component Tests** (rendering, data display, loading states) - **15 points**
+- **Card Component Tests** (rendering, click interactions) - **15 points**
+- **Error Boundary Tests** (error catching, fallback UI, test error button) - **15 points**
+- **API Integration Tests** (mocked API calls, success/error handling) - **15 points**
+
+### Penalties
+
+- TypeScript isn't used: **-95 points**
```

Comment:
We don't mention Husky integration with tests neither in points, not in penalties, I suggest to add penalty for that if missing